shared/src/commonMain/kotlin/org/hyperskill/app/study_plan/widget/presentation/StudyPlanWidgetFeature.kt

```diff
@@ -82,20 +73,18 @@ object StudyPlanWidgetFeature {
         object StageImplementUnsupportedModalHiddenEventMessage : Message
     }
 
-    internal sealed interface StudyPlanFetchResult : Message {
-        data class Success(
-            val studyPlan: StudyPlan,
-            val attemptNumber: Int,
-            val showLoadingIndicators: Boolean
-        ) : StudyPlanFetchResult
+    internal sealed interface InternalMessage : Message {
+        data class Initialize(val forceUpdate: Boolean = false) : InternalMessage
 
-        object Failed : StudyPlanFetchResult
-    }
+        object ReloadContentInBackground : InternalMessage
 
-    internal sealed interface SectionsFetchResult : Message {
-        data class Success(val sections: List<StudyPlanSection>) : SectionsFetchResult
+        object FetchLearningActivitiesWithSectionsFailed : InternalMessage
+        data class FetchLearningActivitiesWithSectionsSuccess(
+            val learningActivities: List<LearningActivity>,
+            val studyPlanSections: List<StudyPlanSection>
+        ) : InternalMessage
```

Comment:
I suggest to make it a sealed class, like `LearningActivitiesFetchResult` to make all messages uniform